src/ArduinoJson/Strings/StringTraits.hpp

```diff
@@ -39,16 +39,27 @@ struct has_data<T,
     : true_type {};
 
 // size_t length() const
+// unsigned int length() const
 // - String
 
 template <class T, class = void>
 struct has_length : false_type {};
 
+template <typename T, typename... Rest>
+struct is_any : false_type {};
+
+template <typename T, typename First>
+struct is_any<T, First> : is_same<T, First> {};
+
+template <typename T, typename First, typename... Rest>
+struct is_any<T, First, Rest...>
+    : integral_constant<bool, is_same<T, First>::value ||
+                                  is_any<T, Rest...>::value> {};
+
 template <class T>
 struct has_length<
-    T, typename enable_if<
-           is_same<decltype(declval<const T>().length()), size_t>::value>::type>
-    : true_type {};
+    T, typename enable_if<is_any<decltype(declval<const T>().length()), size_t,
```

Comment:
Please remove `is_any` and use `is_unsigned`.

Reply:
OK :+1: done